src/main/java/ui/GUIController.java

```diff
@@ -25,7 +25,10 @@
     private final PanelControl panelControl;
     private final UI ui;
     private final Label apiBox;
+    private int prevRemainingRequests = 0;
+    private int lastApiCallsUsed = 0;
     private String defaultRepoId;
```

Comment:
These fields can have some comments explaining their purpose since it's not very obvious from their names here. Also the variables can be a bit clearer names such as `previousRemainingApiRequests`, and `lastNumberOfApiCallsUsed`.